app/bootstrap.js

```diff
@@ -38,6 +38,78 @@ function getOption(name) {
   return _CONFIG_DATA[name] || '';
 }
 
+/**
+ * Hide the loading indicator once the app is fully loaded
+ */
+function hideAppLoadingIndicator() {
+  const indicator = document.getElementById('jupyterlite-loading-indicator');
+  if (indicator) {
+    indicator.classList.add('hidden');
+    indicator.addEventListener('animationend', () => {
+      indicator.remove();
+      // Remove theme classes after the loading indicator is removed
+      document.body.classList.remove('jp-mod-dark', 'jp-mod-light');
+    }, { once: true });
+  }
+}
+
+/**
+ * Apply theme to loading indicator based on saved settings in IndexedDB
+ */
+async function applyThemeToAppLoadingIndicator() {
+  const indicator = document.getElementById('jupyterlite-loading-indicator');
+  if (!indicator) {
+    return; // Not on lab page, no need to apply theme
+  }
+
+  try {
+    const baseUrl = getOption('baseUrl');
+    const defaultStorageName = `JupyterLite Storage - ${baseUrl}`;
+    const storageName = getOption('settingsStorageName') || defaultStorageName;
+
+    const localforageModule = await import('localforage');
+    const localforage = localforageModule.default;
+
+    const settingsDB = localforage.createInstance({
+      name: storageName,
+      storeName: 'settings'
+    });
+
+    const key = '@jupyterlab/apputils-extension:themes';
+    const settings = await settingsDB.getItem(key);
+
+    let isDarkTheme = false;
+
+    if (settings) {
+      // Use regex to find "theme": "name of theme" pattern, since the
+      // settings are stored as a raw string
+      const themeRegex = /"theme"\s*:\s*"([^"]+)"/i;
+      const matches = settings.match(themeRegex);
+
+      if (matches && matches[1]) {
+        const themeName = matches[1].toLowerCase();
+        isDarkTheme =
+          themeName.includes('dark') ||
+          themeName.includes('night') ||
+          themeName.includes('black');
+      }
+    }
+
+    document.body.classList.remove('jp-mod-dark', 'jp-mod-light');
+
+    if (isDarkTheme) {
+      document.body.classList.add('jp-mod-dark');
+    } else {
+      document.body.classList.add('jp-mod-light');
+    }
+  } catch (e) {
+    console.warn('Could not apply theme to loading indicator:', e);
+    // Fallback to light theme on error
+    document.body.classList.remove('jp-mod-dark');
+    document.body.classList.add('jp-mod-light');
+  }
+}
```

Comment:
Perhaps we could sniff the browser settings by checking for the `data-theme` and `data-mode` HTML attributes if they exist, and use them to determine the theme instead of falling back to a light theme by default?

Reply:
You mean `data-jp-theme` on `body`?

![image](https://github.com/user-attachments/assets/b5f214fe-8ed1-4c23-99a4-9af7776524fc)

I think these attributes are set after the application has started, so they are likely added a bit too late in the bootstrap process.

Reply:
Ah right, but those attributes will be in the parent window, and will need to be "sent" to the IFrame. Maybe https://github.com/jupyterlite/jupyterlite/issues/1637 could help define custom logic to get these attributes.